src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement/HostedServices/Services/AltinnAdminRoleSyncService.cs

```diff
@@ -0,0 +1,169 @@
+                            
+                            int revokes = await _assignmentService.RevokeAdminAssignmentPackages(
+                                item.ToUserPartyUuid.Value,
+                                item.FromPartyUuid,
+                                packageUrns,
+                                options,
+                                cancellationToken);
+                            
+                            if (revokes == 0)
+                            {
+                                _logger.LogWarning(
+                                    "Failed to delete assignmentpackages for FromParty: {FromParty}, ToParty: {ToParty}, PackageUrns: {packageUrn}",
+                                    item.FromPartyUuid,
+                                    item.ToUserPartyUuid,
+                                    string.Join(", ", packageUrns));
+                            }
+                        }
+                        else
+                        {
+                            if (item.ToUserPartyUuid == null)
+                            {
+                                _logger.LogWarning(
+                                    "The delegation is missing ToUserPartyUuid so it is not a valid admin delegation {FromParty}, ToParty: {ToParty}, PackageUrns: {PackageUrn}",
+                                    item.FromPartyUuid,
+                                    item.ToUserPartyUuid,
+                                    string.Join(", ", packageUrns));
+                                continue;
+                            }
+
+                            int adds = await _assignmentService.ImportAdminAssignmentPackages(item.ToUserPartyUuid.Value, item.FromPartyUuid, packageUrns, options, cancellationToken);
+                            if (adds == 0)
+                            {
+                                _logger.LogWarning(
+                                    "Failed to delete delegation for FromParty: {FromParty}, ToParty: {ToParty}, PackageUrns: {packageUrn}",
+                                    item.FromPartyUuid,
+                                    item.ToUserPartyUuid,
+                                    string.Join(", ", packageUrns));
+                            }
+                        }
+
+                    }
+                }
+
+                if (string.IsNullOrEmpty(page?.Content?.Links?.Next))
+                {
+                    return;
+                }
+
+                await UpdateLease(ls, data => data.AltinnAdminRoleStreamNextPageLink = page.Content.Links.Next, cancellationToken);
+            }
+        }
+
+        private List<string> GetAdminPackageFromRoleTypeCode(string roleTypeCode, CancellationToken cancellationToken = default)
+        {
+            List<string> packages = new List<string>();
+            
+            switch (roleTypeCode)
+            {
+                case "ADMAI":
+                    packages.Add("urn:altinn:accesspackage:tilgangsstyrer");
```

Comment:
Har vel vært diskutert litt frem å tilbake også om ADMAI rollen skal automatisk gi tilgangsstyrer pakken.
Eller om vi skal tvinge at den evt. må delegeres manuelt skal for de som skal beholde tilgangen etter at vi skrur av A2-rollene.. så må vel avklares dette også.